source/material_model/grain_size.cc

```diff
@@ -110,6 +110,25 @@ namespace aspect
     }
 
 
+    template <int dim>
+    void
+    GrainSize<dim>::update()
+    {
+      data.initial_time = 0.0;
+      data.final_time = this->get_timestep();
+      data.initial_step_size = this->get_timestep();
+      data.output_period = this->get_timestep();
+      data.minimum_step_size = 1.e-6*this->get_timestep();
+      data.maximum_order = 3;
+      data.maximum_non_linear_iterations = 30;
+
+      // Because both tolerances are added, we set the absolute
+      // tolerance to 0.
+      data.relative_tolerance = 1e-3;
+      data.absolute_tolerance = 0;
```

Comment:
I would move all of this right above the place where you now create the ARKode solver object.